### DISCUSSION TITLE
Is there a way to use basic auth to get the user's token?

### DISCUSSION THREAD

**Discussion by Lon1** [netbox-community/netbox#5832] [Q&A] [answered]
I think the title says it all.  Like in Apigee, a user must use basic auth first, in order to the token back.  Can requests / pynetbox do the same? Example?

Thanks!

**Comment by candlerb** [comment, marked as answer]
Netbox doesn't issue tokens programatically.  You would have to write your own token-issuing service.

Alternatively, you can put a proxy in front of Netbox, and apply whatever authentication you like: use [REMOTE_AUTH](https://netbox.readthedocs.io/en/stable/configuration/optional-settings/#remote_auth_enabled) for the proxy to use a header to forward the authenticated ID to Netbox.
